clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -609,14 +686,15 @@ public synchronized void handleCompletedBatch(ProducerBatch batch, ProduceRespon
     }
 
     public synchronized void transitionToUninitialized(RuntimeException exception) {
-        transitionTo(State.UNINITIALIZED);
+        transitionTo(State.UNINITIALIZED, exception, InvalidStateDetectionStrategy.BACKGROUND);
         if (pendingTransition != null) {
             pendingTransition.result.fail(exception);
         }
         lastError = null;
     }
 
-    public synchronized void maybeTransitionToErrorState(RuntimeException exception) {
+    public synchronized void maybeTransitionToErrorState(RuntimeException exception,
```

Comment:
Just to double check: the transitionToFatalError call on line 702 does not use the invalidStateDetectionStrategy param because all of the exceptions in the condition come from a server response, thus making it a BACKGROUND type call?

Reply:
Correct. However, I went ahead and made the parameter explicit to avoid confusion.